compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -58,8 +58,8 @@ class Compiler {
          new CheckReentrant,         // Internal use only: Check that compiled program has no data races involving global vars
          new ElimPackagePrefixes,    // Eliminate references to package prefixes in Select nodes
          new CookComments,           // Cook the comments: expand variables, doc, etc.
-         new CompleteJavaEnums) ::   // Fill in constructors for Java enums
-    List(new CheckStatic,            // Check restrictions that apply to @static members
+         new CheckStatic) ::         // Check restrictions that apply to @static members
+    List(new CompleteJavaEnums,      // Fill in constructors for Java enums
```

Comment:
Why is this swap needed? Since we are adding the `@static` annotation in `CompleteJavaEnums`, should this phase not go before `CheckStatic` so that the latter can check the newly added static members?

Reply:
`CheckStatic` will reject our code, because it requires that a `@static` member is located in a companion object, not the class.

Reply:
Would it be a better idea to modify `CheckStatic` to accept the code? E.g., "1. Only objects and classes that extend java.lang.Enum can have members annotated with `@static`"?

The meaning of `CheckStatic` is to provide guarantees on `@static`. If there exists code that doesn't follow the existing guarantees, this complicates the codebase maintenance and usage. We have the exact same scenario with setting flags on completion: there are rules about which flags are immutable wrt completion but they are not enforced. And hence there are lots of places in the codebase that don't follow these rules.

Reply:
I have a commit message in f6f3b28cfcc1f79d002a8af950ef49a84bc88b07: the `@static` infrastructure needs fundamentla refactoring, but it's out of the scope of this PR.